framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2104,7 +2171,7 @@ public boolean shouldSuppressWarnings(Tree tree, String errKey) {
         }
 
         // trees.getPath might be slow, but this is only used in error reporting
-        @Nullable TreePath path = trees.getPath(this.currentRoot, tree);
+        @Nullable TreePath path = getTreePathCacher().getPath(currentRoot, tree);
```

Comment:
At the very bottom of this file is method `getPathToCompilationUnit`. Could it's implementation also benefit from using the `TreePathCacher`?

Reply:
Thanks for the reminder! Yes, I have updated the method.